flow/project.py

```diff
@@ -384,7 +383,7 @@ def __repr__(self):
         )
 
     def __hash__(self):
-        return int(sha1(self.id.encode("utf-8")).hexdigest(), 16)
+        return hash(self.id)
```

Comment:
The `__hash__` method here does not require a function like md5 or sha1. It's sufficient to just return `hash(self.id)`. This appeared to be a legacy pattern that was copied for no apparent reason, and it is the only place left in signac or signac-flow where such a custom hash function is provided.

The purpose of Python's `hash` is to help with hash-based data structures like `dict` and `set` and (potentially) to be used as a required condition for object equality. The requirements of md5 / sha1 are stricter and they are much more expensive to compute than the built-in `hash` function.

Reply:
Sorry, I left this comment without appropriate context. I was primarily responding to @kidrahahjo, and indicating future work.

I agree that using `hash` is more appropriate here. What I was discussing is the use of sha1 in general w.r.t. security as per the original comment, because I noticed that sha1 is still imported by this file but md5 was added. I think that in general we should use md5 everywhere because that's our historical usage. If we want to change to something else (aggregation uses sha1 in a couple places) that should be a discussion and be done globally. In general we don't have security concerns; we're not storing passwords, or any encrypted data, we just want something reasonably collision-free. If we start seeing collisions and need a better algorithm, that's a schema change in signac and may lead to breaking changes between signac-flow releases (because an aggregate's name changes), so I think we should be consistent for now and make uniform changes as needed later.